noria/src/data.rs

```diff
@@ -561,32 +562,25 @@ impl<'a> From<&'a str> for DataType {
     }
 }
 
-impl TryFrom<mysql::Value> for DataType {
+impl TryFrom<Value> for DataType {
     type Error = &'static str;
 
-    fn try_from(v: mysql::Value) -> Result<Self, Self::Error> {
+    fn try_from(v: Value) -> Result<Self, Self::Error> {
         match v {
-            mysql::Value::NULL => Ok(DataType::None),
-            mysql::Value::Bytes(v) => {
-                let len = v.len();
-                if len <= TINYTEXT_WIDTH {
-                    let mut bytes = [0; TINYTEXT_WIDTH];
-                    for (i, c) in v.iter().enumerate() {
-                        bytes[i] = *c;
-                    }
-
-                    Ok(DataType::TinyText(bytes))
-                } else {
-                    match ArcCStr::try_from(&v[..]) {
-                        Ok(arc_c_str) => Ok(DataType::Text(arc_c_str)),
-                        Err(_) => Err("std::ffi::FromBytesWithNulError"),
-                    }
+            Value::NULL => Ok(DataType::None),
+            Value::Bytes(v) => {
+                if v.iter().any(|b| b == &0) {
```

Comment:
We don't have code to deal with exactly this conversion (`Vec<u8>` -> `DataType`) somewhere already? I feel like I've written it in the past...

Reply:
There is no conversion. The closest we have is from `String` or from `&str`.

Reply:
error[E0277]: the trait bound `data::DataType: std::convert::From<std::vec::Vec<u8>>` is not satisfied
   --> noria/src/data.rs:572:35
    |
572 |             Value::Bytes(v) => Ok(v.into()),
    |                                   ^^^^^^^^ the trait `std::convert::From<std::vec::Vec<u8>>` is not implemented for `data::DataType`

Reply:
Actually, I suppose the reason we don't have that impl is that the general conversation from `Bytes` to `String` just isn't safe. If the bytes aren't a valid utf-8, or contain a NULL-byte, then they cannot be represented in Noria. Arguably we should have a separate `Bytes` variant, but it's probably not worth adding at the moment. I guess this will have to do for now. Although the checking does make the conversion pretty expensive :thinking: